src/cdef_apply.rs

```diff
@@ -107,24 +105,28 @@ unsafe fn backup2x8<BD: BitDepth>(
     layout: Rav1dPixelLayout,
     flag: Backup2x8Flags,
 ) {
-    let mut y_off: ptrdiff_t = 0 as c_int as ptrdiff_t;
+    let mut y_off = 0;
+
     if flag.contains(Backup2x8Flags::Y) {
         for y in 0..8 {
             BD::pixel_copy(
                 &mut dst[0][y],
-                slice::from_raw_parts(&mut *src[0].offset(y_off + x_off as isize - 2), 2),
+                slice::from_raw_parts(src[0].offset(y_off + x_off as isize - 2), 2),
```

Comment:
is that `&mut` something that `c2rust` inserts? it's not really needed and also unsound when that pointer is dangling (a dangling pointer would be sound if the slice length were 0). Not a problem in practice here but in general it seems odd to me.

Reply:
Somewhat (it also looks extra weird since we didn't delete it as we partially cleaned it up/bitdepth deduplicated it).  `c2rust transpile` sometimes has redundant `&mut *`s, and it also generates that kind of code when it really should generate `addr_of_mut!` to be sound (see https://github.com/immunant/c2rust/issues/301).